File: src/Entity/Artiste.php
<?php

namespace App\Entity;

use App\Repository\ArtisteRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: ArtisteRepository::class)]
class Artiste extends User
{

    #[ORM\Column(type: Types::DATE_IMMUTABLE, nullable: true)]
    #[Assert\LessThanOrEqual('today', message: 'Birth date cannot be in the future.')]
    private ?\DateTimeImmutable $birthDate = null;

    #[ORM\Column(type: Types::TEXT, nullable: true)]
    private ?string $bio = null;

    #[ORM\Column(length: 255, nullable: true)]
    #[Assert\Length(max: 255, maxMessage: 'Styles cannot be longer than {{ limit }} characters.')]
    private ?string $styles = null;

    #[ORM\Column(length: 255, nullable: true)]
    #[Assert\Length(max: 255, maxMessage: 'Facebook URL cannot be longer than {{ limit }} characters.')]
    #[Assert\Url(message: 'Please enter a valid Facebook URL.')]
    private ?string $facebook = null;

    #[ORM\Column(length: 255, nullable: true)]
    #[Assert\Length(max: 255, maxMessage: 'Instagram URL cannot be longer than {{ limit }} characters.')]
    #[Assert\Url(message: 'Please enter a valid Instagram URL.')]
    private ?string $instagram = null;

    #[ORM\Column(length: 255, nullable: true)]
    #[Assert\Length(max: 255, maxMessage: 'Portfolio URL cannot be longer than {{ limit }} characters.')]
    #[Assert\Url(message: 'Please enter a valid portfolio URL.')]
    private ?string $portfolioUrl = null;

    #[ORM\Column(options: ['default' => false])]
    private ?bool $verified = false;

    public function getBio(): ?string
    {
        return $this->bio;
    }

    public function setBio(?string $bio): static
    {
        $this->bio = $bio;

        return $this;
    }

    public function getStyles(): ?string
    {
        return $this->styles;
    }

    public function setStyles(?string $styles): static
    {
        $this->styles = $styles;

        return $this;
    }

    public function getFacebook(): ?string
    {
        return $this->facebook;
    }

    public function setFacebook(?string $facebook): static
    {
        $this->facebook = $facebook;

        return $this;
    }

    public function getInstagram(): ?string
    {
        return $this->instagram;
    }

    public function setInstagram(?string $instagram): static
    {
        $this->instagram = $instagram;

        return $this;
    }

    public function getPortfolioUrl(): ?string
    {
        return $this->portfolioUrl;
    }

    public function setPortfolioUrl(?string $portfolioUrl): static
    {
        $this->portfolioUrl = $portfolioUrl;

        return $this;
    }

    public function getBirthDate(): ?\DateTimeImmutable
    {
        return $this->birthDate;
    }

    public function setBirthDate(?\DateTimeImmutable $birthDate): static
    {
        $this->birthDate = $birthDate;

        return $this;
    }

    public function isVerified(): ?bool
    {
        return $this->verified;
    }

    public function setVerified(bool $verified): static
    {
        $this->verified = $verified;

        return $this;
    }
}

File: Java_Application/src/main/java/com/pegasus/entities/Artiste.java
package com.pegasus.entities;

import java.time.LocalDate;
import java.time.LocalDateTime;

public class Artiste extends User {
    private String bio;
    private String styles;
    private String facebook;
    private String instagram;
    private String portfolioUrl;
    private boolean verified;
    private LocalDate birthDate;

    public Artiste() {
    }

    public Artiste(
            String email,
            String roles,
            String password,
            String username,
            String phone,
            String avatarUrl,
            LocalDateTime createdAt,
            String status,
            String dtype,
            String bio,
            String styles,
            String facebook,
            String instagram,
            String portfolioUrl,
            boolean verified,
            LocalDate birthDate
    ) {
        super(
                email,
                roles,
                password,
                username,
                phone,
                avatarUrl,
                createdAt,
                status,
                dtype
        );
        this.bio = bio;
        this.styles = styles;
        this.facebook = facebook;
        this.instagram = instagram;
        this.portfolioUrl = portfolioUrl;
        this.verified = verified;
        this.birthDate = birthDate;
    }

    public Artiste(
            Integer id,
            String email,
            String roles,
            String password,
            String username,
            String phone,
            String avatarUrl,
            LocalDateTime createdAt,
            String status,
            String dtype,
            String bio,
            String styles,
            String facebook,
            String instagram,
            String portfolioUrl,
            boolean verified,
            LocalDate birthDate
    ) {
        this(
                email,
                roles,
                password,
                username,
                phone,
                avatarUrl,
                createdAt,
                status,
                dtype,
                bio,
                styles,
                facebook,
                instagram,
                portfolioUrl,
                verified,
                birthDate
        );
        setId(id);
    }

    public String getBio() {
        return bio;
    }

    public void setBio(String bio) {
        this.bio = bio;
    }

    public String getStyles() {
        return styles;
    }

    public void setStyles(String styles) {
        this.styles = styles;
    }

    public String getFacebook() {
        return facebook;
    }

    public void setFacebook(String facebook) {
        this.facebook = facebook;
    }

    public String getInstagram() {
        return instagram;
    }

    public void setInstagram(String instagram) {
        this.instagram = instagram;
    }

    public String getPortfolioUrl() {
        return portfolioUrl;
    }

    public void setPortfolioUrl(String portfolioUrl) {
        this.portfolioUrl = portfolioUrl;
    }

    public boolean isVerified() {
        return verified;
    }

    public void setVerified(boolean verified) {
        this.verified = verified;
    }

    public LocalDate getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(LocalDate birthDate) {
        this.birthDate = birthDate;
    }

    @Override
    public String toString() {
        return "Artiste{user=" + super.toString()
                + ", bio='" + bio + "'"
                + ", styles='" + styles + "'"
                + ", facebook='" + facebook + "'"
                + ", instagram='" + instagram + "'"
                + ", portfolioUrl='" + portfolioUrl + "'"
                + ", verified=" + verified
                + ", birthDate=" + birthDate
                + "}";
    }
}
